libdnf/module/module_sack.cpp

```diff
@@ -145,6 +149,150 @@ BaseWeakPtr ModuleSack::get_base() const {
     return p_impl->base;
 }
 
+std::tuple<
+    std::vector<std::string>,
+    std::vector<std::string>,
+    std::vector<std::string>,
+    std::vector<std::string>,
+    rpm::ReldepList>
+ModuleSack::Impl::collect_data_for_modular_filtering() {
+    base->get_module_sack()->add_modules_without_static_context();
+    auto & allPackages = get_modules();
+    // TODO(jmracek) Add support of demodularized RPMs
+    // auto demodularizedNames = getDemodularizedRpms(modulePackageContainer, allPackages);
+
+    std::vector<std::string> include_NEVRAs;
+    std::vector<std::string> exclude_NEVRAs;
+    std::vector<std::string> names;
+    std::vector<std::string> src_names;
+    libdnf::rpm::ReldepList reldep_name_list(base);
+    for (const auto & module : allPackages) {
+        auto artifacts = module->get_artifacts();
+        if (module->is_active()) {
+            std::string packageID{module->get_name_stream()};
+            packageID.append(".");
+            packageID.append(module->get_arch());
+            //auto it = demodularizedNames.find(packageID);
+            //if (it == demodularizedNames.end()) {
+            if (true) {
+                for (const auto & rpm : artifacts) {
+                    auto nevras = rpm::Nevra::parse(rpm, {rpm::Nevra::Form::NEVRA});
+                    if (nevras.empty()) {
+                        // TODO(jmracek) Unparsable NEVRA - What to do?
+                        continue;
+                    }
+                    auto & nevra = *nevras.begin();
+                    auto arch = nevra.get_arch();
+                    if (arch == "src" || arch == "nosrc") {
+                        src_names.push_back(nevras.begin()->get_name());
+                    } else {
+                        names.push_back(nevra.get_name());
+                        reldep_name_list.add_reldep(nevra.get_name());
+                    }
+                }
+            }
+            // } else {
+            //     for (const auto &rpm : artifacts) {
+            //         auto nevras = rpm::Nevra::parse(rpm, {rpm::Nevra::Form::NEVRA});
+            //         if (nevras.empty()) {
+            //             // TODO(jmracek) Unparsable NEVRA - What to do?
+            //             continue;rpm/package_query.hpp
+            //         }
+            //         auto & nevra = *nevras.begin();
+            //         bool found = false;
+            //         for ( auto & demodularized : it->second) {
+            //             if (nevra.get_name() == demodularized) {
+            //                 found = true;
+            //                 break;
+            //             }
+            //         }
+            //         if (found) {
+            //             continue;
+            //         }
```

Comment:
Could it be simplified like this or would it be less effective?
```
//     if (std::find(it->second.begin(), it->second.end(), nevra.get_name()) != it->second.end()) {
//         continue;
//     }
```
Also, this whole "else" branch is almost the same as the "if" branch, but I suppose it would be slower if the check was inside, hence the redundancy, right?

Reply:
Thank you for note. I would prefer to modify the code during implementation of demodularization. Is it OK for you?

Reply:
Sure, I was just wondering about it as I was trying to understand what's going on there.